src/ngraph/runtime/cpu/cpu_emitter.cpp

```diff
@@ -851,25 +851,70 @@ namespace ngraph
                     }
                 }
 #else
-                auto axis =
-                    (dynamic_cast<const ngraph::op::Concat*>(node))->get_concatenation_axis();
 
-                std::vector<std::string> arg_names;
-                std::vector<Shape> arg_shapes;
-
-                for (auto arg : args)
+                if (runtime::cpu::mkldnn_utils::use_mkldnn_kernel(node))
                 {
-                    arg_names.push_back(arg.get_name());
-                    arg_shapes.push_back(arg.get_shape());
+                    std::vector<mkldnn::memory::format> inputs_format;
+                    std::vector<mkldnn::memory::desc> inputs_data_desc;
+
+                    for (size_t i = 0; i < args.size(); i++)
+                    {
+                        inputs_format.push_back(
+                            runtime::cpu::mkldnn_utils::get_input_mkldnn_format(node, i));
+                    }
+
+                    auto result_format =
+                        runtime::cpu::mkldnn_utils::get_output_mkldnn_format(node, 0);
+                    auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
+                    for (size_t i = 0; i < args.size(); i++)
+                    {
+                        inputs_data_desc.push_back(
+                            mkldnn_emitter->build_memory_descriptor(args[i], inputs_format[i]));
+                    }
+
+                    auto result_desc =
+                        mkldnn_emitter->build_memory_descriptor(out[0], result_format);
+
+                    size_t concat_index = 0;
+                    size_t concat_dims =
```

Comment:
nitpick: `concant_dim` or `axis`? it reads like it's plural...

Reply:
my bad, i'll change it to concat_dim, which is what mkldnn api likes it to call